src/worker/builder.rs

```diff
@@ -281,6 +285,55 @@ impl<E: 'static> WorkerBuilder<E> {
         self
     }
 
+    /// Send resources consumption stats to Faktory.
+    ///
+    /// Faktory currently accepts memory stats (specifically, RSS in kilobytes)
+    /// from workers. This data is then made available in the Web UI. Call this
+    /// method on the builder in order to opt into collecting and sending stats
+    /// to the server.
+    ///
+    /// ```no_run
+    /// # tokio_test::block_on(async {
+    /// let _w = faktory::Worker::builder()
+    ///     .register_fn("jobtype", move |_| async { Ok::<(), std::io::Error>(()) })
+    ///     .with_sysinfo()
+    ///     .connect()
+    ///     .await
+    ///     .unwrap();
+    /// # });
+    /// ```
+    ///
+    /// Note that enabling this behavior might imply that the running program
+    /// may need some extra capabilities, e.g. - for Linix platform - permissions
+    /// to read `/proc/[PID]/statm` file, where PID is your running program's id.
+    /// No system-wide or other processes data gets collected. Furthermore, it is
+    /// only the minimal amount of data (RSS as of Faktory v1.9.3) about the current
+    /// process that gets scraped and sent over to the Faktory server periodically.
+    /// In case the process lacks permissions, this will emit a warning and will
+    /// not install the sysinfo component inside of the worker, i.e. will not be
+    /// sending stats to the Faktory server.
+    ///
+    /// Also note that this does nothing (other than emitting a warning) in case
+    /// the target OS is not supported by the
+    /// [`sysinfo`](https://docs.rs/sysinfo/latest/sysinfo/index.html#supported-oses)
+    /// crate which is being used internally.
+    #[cfg(feature = "sysinfo")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "sysinfo")))]
+    pub fn with_sysinfo(mut self) -> Self {
+        match super::system::System::try_new() {
+            Err(_) => {
+                tracing::warn!("cannot install sysinfo component: the target OS is supported but access to stats files is denied");
```

Comment:
This should almost certainly include the `e` from `Err(e)`, since we don't _know_ that this is due to a permission error.

EDIT: Ah, the error type here is just `()` (for now). Let's then do:

```suggestion
            Err(()) => {
                tracing::warn!("cannot install sysinfo component: the target OS is supported but access to stats failed");
```

So that if we do get richer errors in the future, we'll know to update this code.